### PR TITLE
Feat/k UI 905 add application codes

### PR DESCRIPTION
**Pull request by 4886Talga**
please review the code

### CODE REVIEW

File: server/controllers/mixedWebAndPdfMemosCtrl.js [hunk 1]
```diff
@@ -4,28 +4,32 @@ const log = require('@kth/log')
...
+        miniFileInfo,
+        applicationCode
+      )
     } else {
       pdfMemos[semester] = {
         ...pdfMemos[semester],
         [courseMemoFileName]: {
           courseCode,
           courseMemoFileName,
           ladokRoundIds: [koppsRoundId],
```
Comment (lenaar) [inline server/controllers/mixedWebAndPdfMemosCtrl.js:31]: `ladokRoundIds: [koppsRoundId]`, has to be removed

File: server/controllers/mixedWebAndPdfMemosCtrl.js [hunk 2]
```diff
@@ -158,13 +175,23 @@ async function _formPrioritizedMemosByCourseRounds(courseCode, pdfMemos, webBase
   const miniMemos = {}
   // firstly fetch web-based
   await webBasedMemos.forEach(
-    ({ ladokRoundIds, memoEndPoint, memoCommonLangAbbr, memoName, semester, version, lastChangeDate }) => {
+    ({
+      ladokRoundIds,
```
Comment (lenaar) [inline server/controllers/mixedWebAndPdfMemosCtrl.js:179]: `ladokRoundIds` has to be removed

File: server/controllers/mixedWebAndPdfMemosCtrl.js [hunk 3]
```diff
@@ -158,13 +175,23 @@ async function _formPrioritizedMemosByCourseRounds(courseCode, pdfMemos, webBase
...
+      version,
+      lastChangeDate,
+      applicationCodes,
+    }) => {
       if (!semester) return
       if (!miniMemos[semester]) miniMemos[semester] = {}
       ladokRoundIds.forEach(roundId => {
         miniMemos[semester][roundId] = {
           courseCode,
           ladokRoundIds,
```
Comment (lenaar) [inline server/controllers/mixedWebAndPdfMemosCtrl.js:193]: `ladokRoundIds` has to be removed

File: server/controllers/mixedWebAndPdfMemosCtrl.js [hunk 4]
```diff
@@ -194,6 +222,7 @@ async function _formPrioritizedMemosByCourseRounds(courseCode, pdfMemos, webBase
         miniMemos[semester][koppsRoundId] = {
           courseCode,
           courseMemoFileName,
+          applicationCodes: [applicationCode],
           ladokRoundIds: [koppsRoundId],
```
Comment (lenaar) [inline server/controllers/mixedWebAndPdfMemosCtrl.js:226]: `ladokRoundIds` has to be removed

File: server/controllers/storedMemoPdfsCtrl.js [hunk 5]
```diff
@@ -34,6 +34,7 @@ async function getStoredMemoPdfListByCourseCode(req, res) {
           courseCode: dbPdfMemo.courseCode,
           pdfMemoUploadDate: dbPdfMemo.pdfMemoUploadDate,
           koppsRoundId: dbPdfMemo.koppsRoundId,
```
Comment (lenaar) [inline server/controllers/storedMemoPdfsCtrl.js:36]: `koppsRoundId` has to be removed

File: server/lib/dbSeveralDocument.js [hunk 6]
```diff
@@ -83,9 +87,20 @@ async function _getSortedMiniMemosForAllYears(courseCode, memoStatus = 'publishe
   logInCaseOfPossibleLimit(webBasedMemos, matchingParameters)
 
   const publishedForAllYears = webBasedMemos.map(dbMemo => {
-    const { _id: memoId, semester, status, ladokRoundIds, memoEndPoint, memoName, memoCommonLangAbbr, version } = dbMemo
+    const {
+      _id: memoId,
+      semester,
+      status,
+      ladokRoundIds,
```
Comment (lenaar) [inline server/lib/dbSeveralDocument.js:94]: `ladokRoundIds` has to be removed

File: server/lib/dbSeveralDocument.js [hunk 7]
```diff
@@ -83,9 +87,20 @@ async function _getSortedMiniMemosForAllYears(courseCode, memoStatus = 'publishe
...
+      memoEndPoint,
+      memoName,
+      memoCommonLangAbbr,
+      version,
+      applicationCodes,
+    } = dbMemo
     const miniMemo = {
       ladokRoundIds,
```
Comment (lenaar) [inline server/lib/dbSeveralDocument.js:102]: `ladokRoundIds` has to be removed

File: server/lib/dbSeveralDocument.js [hunk 8]
```diff
@@ -137,13 +152,15 @@ async function getMemosFromPrevSemester(courseCode, fromSemester) {
         semester,
         status,
         ladokRoundIds,
```
Comment (lenaar) [inline server/lib/dbSeveralDocument.js:154]: `ladokRoundIds` has to be removed

File: server/lib/dbSeveralDocument.js [hunk 9]
```diff
@@ -137,13 +152,15 @@ async function getMemosFromPrevSemester(courseCode, fromSemester) {
...
+        applicationCodes,
         memoEndPoint,
         memoName,
         memoCommonLangAbbr,
         version,
       } = dbMemo
       const miniMemo = {
         ladokRoundIds,
```
Comment (lenaar) [inline server/lib/dbSeveralDocument.js:162]: `ladokRoundIds` has to be removed

File: server/models/mainMemoModel.js [hunk 10]
```diff
@@ -26,7 +26,13 @@ const schema = mongoose.Schema({
     type: Array,
     items: String,
     trim: true,
-    required: [true, 'Enter course rounds'],
```
Comment (lenaar) [inline server/models/mainMemoModel.js:29]: `ladokRoundIds` has to be removed

File: test/mocks/draftOfPublished.js [hunk 11]
```diff
@@ -1,6 +1,7 @@
 const draftOfPublished = {
   _id: "ObjectId('5faaa2a5cdd79565130f9854ca')",
   ladokRoundIds: ['1'],
```
Comment (lenaar) [inline test/mocks/draftOfPublished.js:3]: `ladokRoundIds` has to be removed

File: test/mocks/publishedMemo.js [hunk 12]
```diff
@@ -1,6 +1,7 @@
 const publishedMemo = {
   _id: "ObjectId('5f35365s9531ddabe323516c46f')",
   ladokRoundIds: ['1'],
```
Comment (lenaar) [inline test/mocks/publishedMemo.js:3]: `ladokRoundIds` has to be removed

**Review by lenaar — CHANGES_REQUESTED** [review]
`ladokRoundIds`,  `koppsRoundId`, `koppsRoundsId` and all types of `roundIds` has to be removed from the code, because it is will be legacy code. Additionally it will be unclear if the code really working independently or still have dependency on rounds ids

File: server/controllers/mixedWebAndPdfMemosCtrl.js (hunk 1 above)
Comment (Rao-Sabahat-Ali) [reply server/controllers/mixedWebAndPdfMemosCtrl.js]: Done

File: server/controllers/mixedWebAndPdfMemosCtrl.js [hunk 13]
```diff
@@ -55,7 +59,16 @@ async function getPdfAndWebMemosListByCourseCode(req, res) {
     })
 
     webBasedMemos.map(
-      ({ ladokRoundIds, memoEndPoint, memoCommonLangAbbr, memoName, semester, version, lastChangeDate }) => {
+      ({
+        ladokRoundIds,
```
[the remark this answers is not in the document]
Comment (Rao-Sabahat-Ali) [reply server/controllers/mixedWebAndPdfMemosCtrl.js]: Done

File: server/controllers/mixedWebAndPdfMemosCtrl.js (hunk 2 above)
Comment (Rao-Sabahat-Ali) [reply server/controllers/mixedWebAndPdfMemosCtrl.js]: Done

File: server/controllers/mixedWebAndPdfMemosCtrl.js (hunk 3 above)
Comment (Rao-Sabahat-Ali) [reply server/controllers/mixedWebAndPdfMemosCtrl.js]: Done

File: server/controllers/mixedWebAndPdfMemosCtrl.js (hunk 4 above)
Comment (Rao-Sabahat-Ali) [reply server/controllers/mixedWebAndPdfMemosCtrl.js]: Done

File: server/controllers/storedMemoPdfsCtrl.js (hunk 5 above)
Comment (Rao-Sabahat-Ali) [reply server/controllers/storedMemoPdfsCtrl.js]: Done

File: server/lib/dbSeveralDocument.js (hunk 6 above)
Comment (Rao-Sabahat-Ali) [reply server/lib/dbSeveralDocument.js]: Done

File: server/lib/dbSeveralDocument.js (hunk 7 above)
Comment (Rao-Sabahat-Ali) [reply server/lib/dbSeveralDocument.js]: Done

File: server/lib/dbSeveralDocument.js (hunk 8 above)
Comment (Rao-Sabahat-Ali) [reply server/lib/dbSeveralDocument.js]: Done

File: server/lib/dbSeveralDocument.js (hunk 9 above)
Comment (Rao-Sabahat-Ali) [reply server/lib/dbSeveralDocument.js]: Done

File: server/models/mainMemoModel.js (hunk 10 above)
Comment (Rao-Sabahat-Ali) [reply server/models/mainMemoModel.js:29]: Done

File: test/mocks/draftOfPublished.js (hunk 11 above)
Comment (Rao-Sabahat-Ali) [reply test/mocks/draftOfPublished.js]: Done

File: test/mocks/publishedMemo.js (hunk 12 above)
Comment (Rao-Sabahat-Ali) [reply test/mocks/publishedMemo.js]: Done

**Review by lenaar — APPROVED** [review]
Looks good to me

**Pull request merged**